compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -950,7 +951,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                             matches::ArmHasGuard(false),
                             Some((Some(&place), span)),
                         );
-                        unpack!(block = self.place_into_pattern(block, pattern, place, false));
+                        let place_builder = PlaceBuilder::from(local);
```

Comment:
this

Reply:
Discussed with you to about this, so resolving comment